src/helpers/levels.ts

```diff
@@ -46,27 +46,28 @@ export function computeLevelsQuote(
     return { failure: 'below_minimum_amount' };
   }
 
+  let cumulativeBaseAmount = quote.baseAmount;
   for (let i = 1; i < levels.length; i++) {
     const nextLevel = levels[i]!;
-    const nextLevelDepth = nextLevel.level.minus(levels[i - 1]!.level);
+    cumulativeBaseAmount = cumulativeBaseAmount.plus(nextLevel.q);
     const nextLevelQuote = quote.quoteAmount.plus(
-      nextLevelDepth.multipliedBy(nextLevel.price)
+      nextLevel.q.multipliedBy(nextLevel.p)
     );
-    if (reqBaseAmount && reqBaseAmount.lte(nextLevel.level)) {
+    if (reqBaseAmount && reqBaseAmount.lte(cumulativeBaseAmount)) {
       const baseDifference = reqBaseAmount.minus(quote.baseAmount);
       const quoteAmount = quote.quoteAmount.plus(
-        baseDifference.multipliedBy(nextLevel.price)
+        baseDifference.multipliedBy(nextLevel.p)
       );
       return { amount: quoteAmount };
```

Comment:
@sameer-hashflow, I tried to fix the levels computation here. Does this look correct?

Reply:
This looks right. Would be nice to run it against an actual maker and validate that we get good results. Can we run this locally against sativa quotes to validate it?